drivers/spi/spi_max32.c

```diff
@@ -287,7 +344,7 @@ static int spi_max32_transceive(const struct device *dev)
 		spi_context_update_rx(ctx, 1, len);
 	}
 #endif
-
+	spi_spin_unlock(dev, key);
```

Comment:
Unlocking a second time?

Reply:
This was moved before line 346, to not cause issues when interrupt drive